lib/ConfigFileProcessor.js

```diff
@@ -75,6 +80,79 @@ class ConfigFileProcessor {
       done(null, results.pop());
     });
   }
+
+  /**
+   * Reads a JSON configuration file and returns a JavaScript object. Does all the necessary processing and resolves all referenced files.
+   * @param {string} filePath
+   */
+  static readJSONConfigFile(file, cb) {
+    ConfigFileProcessor._readJSONConfigFile(file, 'root', (err, config) => {
+      logger.debug('Final configuration: ' + JSON.stringify(config, null, 4));
+      cb(err, config);
+    });
+  }
+
+  static _readJSONConfigFile(file, fieldPath, cb) {
+    const dirname = path.dirname(file);
+    logger.debug('Reading JSON config from file %s', chalk.cyan(file));
+    return async.waterfall([
+      next => fs.readFile(file, next),
+      (data, next) => {
+        let json = null;
+        try {
+          logger.debug('Parsing JSON config from file: %s', chalk.cyan(file));
+          json = JSON.parse(data);
+        } catch (error) {
+          logger.warn('Invalid JSON in file %s', chalk.cyan(file));
+          return next(error);
+        }
+        return next(null, json);
+      },
+      (json, next) => {
+        logger.debug('Resolving referenced files for %s', chalk.cyan(file));
+        ConfigFileProcessor._processConfigFileValue(json, dirname, fieldPath, (err, processedValue) => {
+          if (err) {
+            logger.warn('Error while resolving referenced files for %s!', chalk.cyan(file));
+            return next(err);
+          }
+
+          logger.debug('Referenced files resolved for %s', chalk.cyan(file));
+
+          return next(null, processedValue);
+        });
+      }
+    ], cb);
+  }
+
+  static _processConfigFileValue(obj, baseFolder, fieldPath, done) {
+    if (Array.isArray(obj) || lodashIsObject(obj)) {
```

Comment:
I think that `isObject` from lodash will return true for an array, although I haven't tested it

Reply:
Good catch. Will change it to isPlainObject.